py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/convert_gtfs_rt.py

```diff
@@ -0,0 +1,162 @@
+    https_mbta_busloc_s3.s3.amazonaws.com_prod_VehiclePositions_enhanced.json.gz
+    https_mbta_integration.mybluemix.net_vehicleCount.gz
+    """
+
+    def __init__(self, config_type: ConfigType) -> None:
+        Converter.__init__(self, config_type)
+
+        # Depending on filename, assign self.details to correct implementation
+        # of GTFSRTDetail class.
+        self.detail: GTFSRTDetail
+
+        if config_type == ConfigType.RT_ALERTS:
+            self.detail = RtAlertsDetail()
+        elif config_type == ConfigType.RT_TRIP_UPDATES:
+            self.detail = RtTripDetail()
+        elif config_type == ConfigType.RT_VEHICLE_POSITIONS:
+            self.detail = RtVehicleDetail()
+        elif config_type == ConfigType.BUS_VEHICLE_POSITIONS:
+            self.detail = RtBusVehicleDetail()
+        elif config_type == ConfigType.BUS_TRIP_UPDATES:
+            self.detail = RtBusTripDetail()
+        else:
+            raise NoImplException(f"No Specialization for {config_type}")
+
+    def convert(self, files: list[str]) -> list[tuple[str, pyarrow.Table]]:
+        pa_table = pyarrow.table(
+            self.detail.empty_table(), schema=self.detail.export_schema
+        )
+
+        logging.info(
+            "Creating pool with %d threads, %d cores available",
+            POOL_SIZE,
+            os.cpu_count(),
+        )
+
+        with ThreadPoolExecutor(max_workers=POOL_SIZE) as executor:
+            for result in executor.map(self.gz_to_pyarrow, files):
+                if result is not None:
+                    pa_table = pyarrow.concat_tables([pa_table, result])
+
+        logging.info(
+            "Completed converting %d files with config %s",
+            len(files),
+            self.config_type,
+        )
+
+        # return a list with a single prefix, table tuple
+        return [(str(self.config_type), pa_table)]
+
+    def record_from_entity(self, entity: dict) -> dict:
+        """
+        Convert an entity in the ingested json dict into a record for a parquet
+        table.
+        """
+
+        def drill_entity(drill_keys: str) -> Optional[dict]:
+            """Util function for recursively getting data out of entity"""
+            ret_dict = entity
+            for key in drill_keys.split(",")[1:]:
+                value = ret_dict.get(key)
+                if value is None:
+                    return value
+                ret_dict = value
+            return ret_dict
+
+        record: dict = {}
+        for drill_keys, fields in self.detail.transformation_schema.items():
+            pull_dict = drill_entity(drill_keys)
+            for get_field in fields:
+                if pull_dict is None:
+                    record[get_field[-1]] = None
+                else:
+                    record[get_field[-1]] = pull_dict.get(get_field[0])
+
+        return record
+
+    def gz_to_pyarrow(self, filename: str) -> Optional[str]:
```

Comment:
pretty much the same. i flattened this out by removing the `_json_to_pyarrow` method. since this is now a member function, we don't have to pass the `Configuration` in and it can update error and archive lists without doing that thing where it returns a table or a string. Now it just optionally returns a table, returning None if something goes wrong.